### DISCUSSION TITLE
[v3] Empty div element on filtering

### DISCUSSION THREAD

**Discussion by addtobag** [livewire/livewire#5964] [1. Help] [answered]
I've got a products component that has a child component for each individual product card and filter component. I'm finding that when i filter, it's leaving an empty div element such as:

`<div wire:id="gMn99ZUBOSo1aQeNXuF1"></div>`

This is then breaking my bootstrap column layout.

Any ideas to what could be causing this?

Thanks

**Comment by nleibbrandt** [comment]
Make sure to add key value for nested components https://livewire.laravel.com/docs/nesting#rendering-children-in-a-loop

**Reply by addtobag** [reply]
Of course, sorry i should have been more specific, child component is included as below with key.

```
@forelse ($products as $product)
                    
    <livewire:store.product.component.card :product='$product' :key="$product->id">
                
@empty

    <div class="col">
        <div class="card mb-3">
            <div class="card-body">
                <h2 class="mb-0 text-center">No Products Found</h2>
            </div>

            </div>
    </div>

@endforelse
```

**Reply by joshhanley** [reply]
@addtobag can you also try adding `wire:key="empty"` to the first div inside the `@empty` tag and let me know how it goes? I suspect though that it may be related to this morph issue PR #5889

**Reply by addtobag** [reply]
Thanks Josh, yes i think it's related, same error message in the Console:

<img width="606" alt="image" src="https://github.com/livewire/livewire/assets/77623201/504e0a42-86c7-4e5a-b042-a62737975931">

**Comment by joshhanley** [comment, marked as answer]
@addtobag can you test this on the new beta.5 release and let me know if you're still having issues? Thanks!

**Reply by addtobag** [reply]
Working great on beta.5 release, thanks Josh!

**Reply by joshhanley** [reply]
@addtobag great, thanks for testing! 🙂
